app/src/main/java/br/com/concrete/magic/presentation/expansions/ExpansionsActivity.kt

```diff
@@ -7,19 +7,22 @@ import android.widget.Toast
 import android.widget.Toast.LENGTH_LONG
 import android.widget.Toast.LENGTH_SHORT
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
 import androidx.databinding.DataBindingUtil
 import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DividerItemDecoration
 import br.com.concrete.magic.R
 import br.com.concrete.magic.databinding.ActivityExpansionsBinding
 import br.com.concrete.magic.utils.isNetworkAvailable
 import kotlinx.android.synthetic.main.activity_expansions.*
 import org.koin.android.viewmodel.ext.android.viewModel
 
+
 class ExpansionsActivity : AppCompatActivity() {
 
     private lateinit var activityExpansionsBinding: ActivityExpansionsBinding
     private var mAdapter: ExpansionsAdapter? = ExpansionsAdapter()
```

Comment:
Por que o adapter está nullable? Ele não pode ser nulo nesse caso, vc ta instanciando. Outro ponto é instanciar o adapter utilizando o `lazy`, dessa forma o adapter só será criado no momento em que for utilizada pela primeira vez.
```suggestion
    private var mAdapter by lazy { ExpansionsAdapter() }
```